src/main/kotlin/game/factory/WarriorFactory.kt

```diff
@@ -0,0 +1,19 @@
+package game.factory
+
+import game.characters.*
+import game.interfaces.BaseWarrior
+import java.util.*
+
+fun  getWarrior(type : String) :  BaseWarrior {
+         var warrior : BaseWarrior? = null
+    if (type.isNotEmpty() and type.isNotBlank()) {
+        when (type.lowercase(Locale.getDefault())) {
+            "knight" -> warrior = Knight(Warrior())
+            "warrior" -> warrior = Warrior()
+
+        }
+    }
+
+    return  warrior!!
```

Comment:
could we make `warrior` late init property to avoid this force unwrap ?

Reply:
This is done. Thank you for the suggestion